lib/hocon/impl/config_concatenation.rb

```diff
@@ -190,4 +215,26 @@ def ignores_fallbacks?
     # for its value.
     false
   end
+
+  def can_equal(other)
+    other.is_a? Hocon::Impl::ConfigConcatenation
+  end
+
+  def ==(other)
+    if other.is_a? Hocon::Impl::ConfigConcatenation
+      can_equal(other) && @pieces == other.pieces
+    else
+      false
+    end
+  end
+
+  def hash
+    @pieces.hash
+  end
```

Comment:
As far as I could tell, `hash` is the built in function objects use to calculate their hashes and whatnot